src/objects/zcl_abapgit_object_enhc.clas.abap

```diff
@@ -91,6 +92,24 @@ CLASS zcl_abapgit_object_enhc IMPLEMENTATION.
     io_xml->read( EXPORTING iv_name = 'LONGTEXT_ID'
                   CHANGING  cg_data = lv_longtext_id ).
 
+    SELECT SINGLE version FROM enhcompheader INTO lv_vers WHERE enhcomposite = ms_item-obj_name.
+    IF sy-subrc = 0.
+      " If object exists already, then set TADIR entry to deleted
+      " otherwise create_enhancement_composite will fail
```

Comment:
I assume it will set it to non-deleted after calling `create_enhancement_composite()`?

Reply:
i tested it. it does.